opencensus/metrics/transport.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import threading
+
+from opencensus.common import utils
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_INTERVAL = 60
+GRACE_PERIOD = 5
+
+
+class TransportError(Exception):
+    pass
+
+
+class PeriodicTask(threading.Thread):
+    """Thread that periodically calls a given function.
+
+    :type func: function
+    :param func: The function to call.
+
+    :type interval: int or float
+    :param interval: Seconds between calls to the function.
+    """
+
+    daemon = True
+
+    def __init__(self, func, interval=None, **kwargs):
+        super(PeriodicTask, self).__init__(**kwargs)
+        if interval is None:
```

Comment:
Curious, what's the difference with `def __init__(self, func, interval=DEFAULT_INTERVAL, **kwargs)`?

Reply:
Just that it makes it easy to make the same argument optional in the calling func.

`interval` comes all the way from `stackdriver.stats_exporter.new_stats_exporter`, if we used `DEFAULT_INTERVAL` here we'd have to either have to use it as the default value in the callers (which means importing it in other modules) or make two separate calls, e.g.

```python
def caller_in_another_module(interval=None):
    ...
    if interval is None:
        task = PeriodicTask()
    else:
        task = PeriodicTask(interval=interval)
```

Using `None` everywhere is convenient, but there are a few big drawbacks: it's less readable since you can't tell the defaults by looking at the signature, you can't change the default value to something other than `None` without changing all the callers too, and it doesn't work for arguments that can actually have `None` as their value.

I'm on the fence here, if you have an opinion on this I'm happy to change it.

Reply:
I don't have strong opinion here. Just observed that using None is a common pattern: https://docs.python-guide.org/writing/gotchas/. Feel free to make your call.